.circleci/config.yml

```diff
@@ -2,20 +2,42 @@ version: 2.1
 
 orbs:
   ft-golang-ci: financial-times/golang-ci@1
+  docker: circleci/docker@1.7.0
+
+jobs:
+  build-and-publish-docker-image:
+          docker:
+            - image: docker:stable-git
+          steps:
+            - checkout
+            - setup_remote_docker
+            - run:
+                name: Build Docker image
+                command: docker build --tag coco/publish-failure-resolver-go:latest .
+            - run:
+                name: Login in Dockerhub
+                command: echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_LOGIN" --password-stdin
+            - deploy:
+                name: Push Docker image
+                command: docker push coco/publish-failure-resolver-go:latest
 
 workflows:
   tests_and_docker:
-    jobs:
+    jobs: 
       - ft-golang-ci/build-and-test:
-          name: build-and-test-project
+          name: build-and-test-project 
 
-      - ft-golang-ci/docker-build:
-          name: build-docker-image
+      - build-and-publish-docker-image:
+          name: build-and-publish-docker-image
           requires:
             - build-and-test-project
+          filters:
+            branches:
+              only: "master"
 
   snyk-scanning:
     jobs:
       - ft-golang-ci/scan:
           name: scan-dependencies
           context: cm-team-snyk
+          
```

Comment:
Formatting inconsistencies could potentially cause issues so let's try to avoid those and fix the empty line @ EOF and the newly added empty spaces on lines 26 and 28.